### PR TITLE
Update to chromium 78.0.3904.87-1

### PR DESCRIPTION
**Pull request by jstkdng**
Also changed some stuff in the PKGBUILD to resemble arch's one, removed uneeded patches and updated one.
I'd like to remove the patches folder altogether but the packages in the AUR are still dependent on it. Will need some input on them.

### CODE REVIEW

File: PKGBUILD [hunk 1]
```diff
@@ -89,18 +98,18 @@ prepare() {
   # Force script incompatible with Python 3 to use /usr/bin/python2
   sed -i '1s|python$|&2|' third_party/dom_distiller_js/protoc_plugins/*.py
 
+  mkdir -p third_party/node/linux/node-linux-x64/bin
+  ln -s /usr/bin/node third_party/node/linux/node-linux-x64/bin/
```
Comment (Eloston) [inline PKGBUILD]: Node.js is only used if we have `optimize_webui` enabled, but we currently have `optimize_webui=false` in ungoogled-chromium's flags.gn. Do you want `optimize_webui` to be enabled?

File: PKGBUILD [hunk 2]
```diff
@@ -124,15 +133,15 @@ build() {
 
   export CC=clang
   export CXX=clang++
-  export AR=llvm-ar
-  export NM=llvm-nm
+  export AR=ar
+  export NM=nm
```
Comment (Eloston) [inline PKGBUILD]: Would this be confused with GCC's `ar` and `nm` instead of LLVM's?

File: PKGBUILD [hunk 3]
```diff
@@ -79,6 +78,16 @@ prepare() {
 
   cd "$srcdir/chromium-${_chromium_version}"
 
+  # Allow building against system libraries in official builds
+  sed -i 's/OFFICIAL_BUILD/GOOGLE_CHROME_BUILD/' \
+    tools/generate_shim_headers/generate_shim_headers.py
+
+  # https://crbug.com/893950
+  sed -i -e 's/\<xmlMalloc\>/malloc/' -e 's/\<xmlFree\>/free/' \
```
Comment (Eloston) [inline PKGBUILD]: This is already handled in `patches/inox-patchset/fix-cfi-failures-with-unbundled-libxml.patch`

File: PKGBUILD [hunk 4]
```diff
@@ -53,10 +53,9 @@ declare -gA _system_libs=(
   [harfbuzz-ng]=harfbuzz
   [icu]=icu
   [libdrm]=
-  [libevent]=libevent
```
Comment (Eloston) [inline PKGBUILD]: Does removing this system dependency have any benefits?

File: PKGBUILD [hunk 5]
```diff
@@ -21,10 +21,10 @@ arch=('x86_64')
...
 license=('BSD')
 depends=('gtk3' 'nss' 'alsa-lib' 'xdg-utils' 'libxss' 'libcups' 'libgcrypt'
-         'ttf-font' 'systemd' 'dbus' 'libpulse' 'pciutils' 'json-glib' 'libva'
-         'desktop-file-utils' 'hicolor-icon-theme' 'jsoncpp' 'openjpeg2')
-makedepends=('python' 'python2' 'gperf' 'yasm' 'mesa' 'ninja' 'git'
-             'clang' 'lld' 'gn' 'llvm' 'quilt')
+         'ttf-font' 'systemd' 'dbus' 'libpulse' 'pciutils' 'json-glib'
+         'desktop-file-utils' 'hicolor-icon-theme')
+makedepends=('python' 'python2' 'gperf' 'yasm' 'mesa' 'ninja' 'nodejs' 'git'
+             'pipewire' 'clang' 'lld' 'gn' 'java-runtime-headless')
```
Comment (Eloston) [inline PKGBUILD]: What are `pipewire` and `java-runtime-headless` dependencies used for?

**Comment by Eloston** [comment]
> I'd like to remove the patches folder altogether but the packages in the AUR are still dependent on it. Will need some input on them.

What are you planning to have instead of a `patches` directory? Do you just want to move the patches up one level?

Also, sorry for breaking your PR. I merged #39 because it was smaller and already ready-to-go.

**Comment by jstkdng** [comment]
>I like removing dependency on Debian's patches, but I'm not sure if some of the PKGBUILD changes are necessary.

I'm trying to be as close as Arch's chromium PKGBUILD, *maybe* they are not necessary to this project but they don't really cause much trouble. Additional build deps will be removed after building, and if some flag is not used, the compiler will just ignore everything related to it.

>What are you planning to have instead of a patches directory? Do you just want to move the patches up one level?

Just have the patches in the same level as the PKGBUILD, and add them manually inside the sources array, how arch is already doing. That would simplyfiy what #34 wanted to achieve. Before it would be tedious since the debian patches still were there. But now it is just the arch patches.

File: PKGBUILD (hunk 4 above)
Comment (jstkdng) [reply PKGBUILD]: We based the PKGBUILD on the inox one, which was some time ago, libevent is no longer included in the third_party folder.

File: PKGBUILD (hunk 3 above)
Comment (jstkdng) [reply PKGBUILD]: I'd like to reduce the amout of patches needed. That whole patch can be replaced with that one liner.

File: PKGBUILD (hunk 2 above)
Comment (jstkdng) [reply PKGBUILD]: Arch uses GCC's ```ar``` and ```nm```, not LLVM's

File: PKGBUILD (hunk 1 above)
Comment (jstkdng) [reply PKGBUILD]: after reading https://chromium.googlesource.com/chromium/src/+/HEAD/docs/optimizing_web_uis.md I don't see a reason to have ```optimize_webui``` false, why did you do that? to reduce compile times?

File: PKGBUILD (hunk 2 above)
Comment (Eloston) [reply PKGBUILD]: Are they also using Clang and LLD? Why would they mix GCC and LLVM compilation tooling like this?

**Comment by Eloston** [comment]
> Just have the patches in the same level as the PKGBUILD, and add them manually inside the sources array, how arch is already doing

This will break our CI checks; how does Arch handle this? Also, this will break the patch updating workflow with `quilt`; does Arch have some alternative workflow?

File: PKGBUILD (hunk 2 above)
Comment (jstkdng) [reply PKGBUILD]: Yes, they are combining clang and gcc stuff, dunno why, tried searching their commits for a reason but nothing appeared, maybe I just didnt look hard enough

Comment (Eloston) [reply PKGBUILD]: Hmm, it's probably fine then. We could try asking the package maintainers too.

**Comment by jstkdng** [comment]
Arch just applies the patches directly, for example, this is what I already have in my repo.
```sh
  msg2 'Applying archlinux patches'

  # Fix VA-API on Intel and Nvidia
  patch -Np1 -i ../vaapi-fix.patch
  
  # Missing include in third_party/blink/public/platform/web_rtc_rtp_source.h
  patch -Np1 -i ../add-missing-include-for-unique_ptr.patch

  # https://crbug.com/957519#c23
  patch -Np1 -i ../dns_util-make-DohUpgradeEntry-non-const.patch

  # https://crbug.com/1005244
  patch -Np1 -i ../fix-shutdown-crash-in-ProfileManager.patch

  # Fixes from Gentoo
  patch -Np1 -i ../chromium-system-icu.patch
  patch -Np1 -i ../chromium-system-zlib.patch

  # https://crbug.com/1005508
  patch -Np1 -i ../fix-spammy-unique-font-matching-log.patch

  # Load Widevine CDM if available
  #patch -Np1 -i ../chromium-widevine.patch

  # https://crbug.com/skia/6663#c10
  patch -Np0 -i ../chromium-skia-harmony.patch

  # Ungoogled chromium stuff
  _ungoogled_repo="$srcdir/$pkgname"
  _utils="${_ungoogled_repo}/utils"
  msg2 'Applying ungoogled chromium patches'
  # Prune binaries
  python "$_utils/prune_binaries.py" ./ "$_ungoogled_repo/pruning.list"
  # Patches themselves
  python "$_utils/patches.py" apply ./ "$_ungoogled_repo/patches"
  # domain substitution
  python "$_utils/domain_substitution.py" apply -r "$_ungoogled_repo/domain_regex.list" -f "$_ungoogled_repo/domain_substitution.list" -c domainsubcache.tar.gz ./
```
Like I said before, I'd like to have the PKGBUILD as close as arch's, but some things stop me from that, like the fact that we have external gn flags whilst arch has its gn flags included in the PKGBUILD, still, I did my best.
>(Also, the merge conflict needs to be resolved)

I'd like to PR my master branch, of course, after fixing the CI stuff, probably keep the ```patches``` folder for compatibility for the AUR.

File: PKGBUILD (hunk 4 above)
Comment (xsmile) [reply PKGBUILD]: On a side note - the libevent library is at `base/third_party/libevent`. With the removal of Debian's system patch, this dependency isn't needed anymore.

**Comment by Eloston** [comment]
> Arch just applies the patches directly, for example, this is what I already have in my repo.

I don't see why it's necessary to keep this aspect of Arch's PKGBUILD considering we already have a nicer `quilt`-based system that makes adding/updating patches easier. Though I'm not an Arch developer, so maybe you guys have some tooling to make that workflow easier. I'll let other people chime in.

**Comment by jstkdng** [comment]
hmm, I'll fix my repo, will PR once it is ready.

**Pull request closed without merging**